src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/tasks/processor/DocumentProcessorTest.java

```diff
@@ -0,0 +1,111 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.tasks.processor;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ScannableItem;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ScannableItemRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.DocumentNotFoundException;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.document.DocumentManagementService;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.document.output.Pdf;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.DocumentProcessor;
+
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.io.Resources.getResource;
+import static com.google.common.io.Resources.toByteArray;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.anyCollection;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DocumentProcessorTest {
+
+    @Mock
+    private ScannableItemRepository scannableItemRepository;
+
+    @Mock
+    private DocumentManagementService documentManagementService;
+
+    private DocumentProcessor documentProcessor;
+
+    @Before
+    public void setup() throws Exception {
+        documentProcessor = new DocumentProcessor(
+            documentManagementService,
+            scannableItemRepository
+        );
+    }
+
+    @Test
+    public void should_update_document_url_when_doc_response_conntains_matching_file_name_and_doc_url()
+        throws Exception {
+        //Given
+        byte[] test1PdfBytes = toByteArray(getResource("test1.pdf"));
+
+        Pdf pdf = new Pdf("test1.pdf", test1PdfBytes);
+        List<Pdf> pdfs = ImmutableList.of(pdf);
+
+        Map<String, String> response = ImmutableMap.of("test1.pdf", "http://localhost/documents/5fef5f98-e875-4084-b115-47188bc9066b");
+
+        when(documentManagementService.uploadDocuments(pdfs)).thenReturn(response);
+
+        //when
+        ScannableItem scannableItem = scannableItem();
+        documentProcessor.processPdfFiles(pdfs, ImmutableList.of(scannableItem));
+
+        //then
+        assertThat(scannableItem.getDocumentUrl()).isEqualTo("http://localhost/documents/5fef5f98-e875-4084-b115-47188bc9066b");
```

Comment:
This assertion forces implementation to modify input arguments and boils down to:

```java
ScannableItem a = new ScannableItem();
a.setDocumentUrl(null);

documentProcessor.processPdfFiles(pdf, ImmutableList.of(scannableItem));

assertThat(a.getDocumentUrl()).isNotNull();
```

Better would be to test if `scannableItemRepository` was asked for saving the item with the URL provided by DM.

Reply:
Good point changes done 👍